### ISSUE TITLE
Concurrent writing and reading hangs event loop

### ISSUE THREAD

**Issue by ns-yrosomakho** [yaa110/tokio-tun#2]
It seems that tokio-tun hangs when trying to write and read concurrently. Is this expected? Is there some pattern that allows concurrent reading and writing?

Here is a minimal example. After 5 seconds reader stops working and packet is actually never sent. Running this example with just reader or just writer (replacing reader loop with sleep) works as expected.

Tokio 1.0.1, Tokio-tun 0.3.1, Rust 1.49 stable.

```rust
use std::net::Ipv4Addr;
use std::os::unix::io::AsRawFd;
use tokio::io::AsyncReadExt;
use tokio::io::AsyncWriteExt;
use tokio_tun::result::Result;
use tokio_tun::TunBuilder;
use tokio::time::{sleep, Duration};


#[tokio::main]
async fn main() -> Result<()> {
    let tun = TunBuilder::new()
        .name("")
        .tap(false)
        .packet_info(false)
        .mtu(1350)
        .up()
        .address(Ipv4Addr::new(10, 0, 0, 1))
        .destination(Ipv4Addr::new(10, 0, 0, 1))
        .broadcast(Ipv4Addr::BROADCAST)
        .netmask(Ipv4Addr::new(255, 0, 0, 0))
        .try_build()?;

    println!("-----------");
    println!("tun created");
    println!("-----------");

    println!(
        "┌ name: {}\n├ fd: {}\n├ mtu: {}\n├ flags: {}\n├ address: {}\n├ destination: {}\n├ broadcast: {}\n└ netmask: {}",
        tun.name(),
        tun.as_raw_fd(),
        tun.mtu().unwrap(),
        tun.flags().unwrap(),
        tun.address().unwrap(),
        tun.destination().unwrap(),
        tun.broadcast().unwrap(),
        tun.netmask().unwrap(),
    );

    println!("---------------------");
    println!("ping 10.1.0.2 to test");
    println!("---------------------");

    let (mut reader, mut writer) = tokio::io::split(tun);

    tokio::spawn( async move {
        sleep(Duration::from_secs(5)).await;
        let packet = [69, 0, 0, 64, 0, 0, 64, 0, 64, 6, 112, 21, 10, 0, 0, 2, 3, 209, 188, 208, 114, 4, 1, 187, 232, 33, 115, 193, 0, 0, 0, 0, 176, 2, 255, 255, 15, 234, 0, 0, 2, 4, 5, 180, 1, 3, 3, 6, 1, 1, 8, 10, 62, 53, 78, 151, 0, 0, 0, 0, 4, 2, 0, 0];
        println!("Sending packet!");
        writer.write(&packet).await.expect("Error!");
        println!("Packet sent");
    });

    let mut buf = [0u8; 1024];
    loop {
        let n = reader.read(&mut buf).await?;
        println!("reading {} bytes: {:?}", n, &buf[..n]);
    }
}
```

**Comment by ns-yrosomakho** [comment]
Btw, stealing raw fd and wrapping it around tokio-fd resolved the issue.
Would be great to implement relevant tokio-fd logic here.

**Comment by yaa110** [comment]
thanks for reporting the issue. Fixed in [`0.3.2`](https://github.com/yaa110/tokio-tun/releases/tag/0.3.2).
